examples/ex301_solidmechanics.jl

```diff
@@ -0,0 +1,105 @@
+# ------------------------------------------------------------------------------
+# solid mechanics example
+# ------------------------------------------------------------------------------
+
+using LinearAlgebra
+
+# setup
+mesh = Mesh3D(1.0, 1.0, 1.0)
+finep = 3
+coarsep = 2
+dimension = 3
+finebasis = TensorH1LagrangeBasis(finep, finep, dimension)
+coarsebasis = TensorH1LagrangeBasis(coarsep, finep, dimension)
+ctofbasis = TensorH1LagrangeBasis(coarsep, finep, dimension, lagrangequadrature = true)
+
+# constants
+e = 1E6                     # Young's modulus
+ν = 0.3                     # Poisson's ratio
+K = e/(3*(1 - 2*ν))         # bulk modulus
+λ = e*ν/((1 + ν)*(1 - 2*ν)) # Lamé parameters
+μ = e/(2*(1 + ν))
+
+# state
+gradu = zeros(3, 3);
+gradu[1, 1:3] = 1*ones(3);
+gradu[2, 1:3] = 2*ones(3);
+gradu[3, 1:3] = 3*ones(3);
+
+function neohookeanweakform(
```

Comment:
@jedbrown, this interface still has some rough spots, I'm really liking how straightforward this was to put together. This is the linearization from our solids code in my LFAToolkit.